java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S2077_java.html

```diff
@@ -1,5 +1,5 @@
 <p>Formatted SQL queries can be difficult to maintain, debug and can increase the risk of SQL injection when concatenating untrusted values into the
-query. However, this rule doesn’t detect SQL injections (unlike rule {rule:javasecurity:S3649}), the goal is only to highlight complex/formatted queries.</p>
+query. However, this rule doesn’t detect SQL injections (unlike rule {rule:java:S3649}), the goal is only to highlight complex/formatted queries.</p>
```

Comment:
Not sure why this got changed, but I think the correct key for this rule is `javasecurity:S3649`, as changing it to java will not link to any rule: https://peach.sonarsource.com/coding_rules?rule_key=java%3AS3649